shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1625,10 +1632,13 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
   NSUInteger start = ((FlutterTextPosition*)range.start).index;
   NSUInteger end = ((FlutterTextPosition*)range.end).index;
   if (_markedTextRange != nil) {
+    UIView* hostView = _textInputPlugin.hostView;
+    NSAssert(hostView == nil || [self isDescendantOfView:hostView], @"%@ is not a descendant of %@",
+             self, hostView);
     // The candidates view can't be shown if the framework has not sent the
     // first caret rect.
     if (CGRectEqualToRect(kInvalidFirstRect, _markedRect)) {
-      return kInvalidFirstRect;
+      return hostView ? [hostView convertRect:kInvalidFirstRect toView:self] : kInvalidFirstRect;
```

Comment:
@LongCatIsLooong I'm a bit iffy on this one and could use another pair of eyes: 

My understanding of [your previous PR](https://github.com/flutter/engine/pull/36707) is that this should return rects in FlutterTextInputView's coord space - not only valid rects, but also this dummy invalid rect (because otherwise an invalid rect could very well become a valid rect with some transforms of FlutterTextInputView).

Reply:
Yeah I think `kInvalidFirstRect` is probably specified in the window's coordinate system so it needs to be transformed. The input view used to be placed at the top left corner of the window iirc.